ui/app/src/modules/System/Sites/Create/BlueprintForm.tsx

```diff
@@ -197,8 +198,9 @@ function BlueprintForm(props: BlueprintFormProps) {
   }
 
   function getSiteId(siteName: string): string {
-    let siteId = siteName.replace(/[^a-zA-Z0-9\s]/g, '').toLowerCase();
-    siteId = siteId.replace(/\s/g, '-');
+    let siteId = siteName.replace(/[^a-zA-Z0-9_\s]/g, '')
+      .replace(/[_\s]/g, '-')
+      .toLowerCase();
```

Comment:
the first regex is stripping off the underscores, so the second won't do anything

Reply:
that regex starts with `^`, so it will actually strip everything that is not `a-zA-Z0-9-_`